frontend/src/app/spindle/spindle.component.ts

```diff
@@ -88,26 +127,30 @@ export class SpindleComponent implements OnInit, OnDestroy {
             });
     }
 
+    downloadAsFile(textData: string, extension: 'tex' | 'json'): void {
+        const fileName = "spindleParseResult." + extension;
+        const blob = new Blob([textData], { type: `application/${extension}` });
+        const url = window.URL.createObjectURL(blob);
+        this.downloadFile(fileName, url);
+    }
+
     ngOnDestroy(): void {
         this.destroy$.next();
         this.destroy$.complete();
     }
 
-    /**
-     * Creates an anchor element `<a></a>` with
-     * the base64 pdf source and a filename with the
-     * HTML5 `download` attribute then clicks on it.
-     * @param  {string} base64 The base64 pdf source
-     * @return {void}
-     */
-    private downloadPDF(base64: string): void {
-        const linkSource = `data:application/pdf;base64,${base64}`;
-        const downloadLink = document.createElement("a");
-        const fileName = "spindleParseResult.pdf";
+    private clearResults(): void {
+        this.term = null;
+        this.textOutput = null;
+        this.lexicalPhrases = [];
+    }
 
-        downloadLink.href = linkSource;
-        downloadLink.download = fileName;
-        downloadLink.click();
-        downloadLink.remove();
+    private downloadFile(fileName: string, url: string): void {
+        const link = document.createElement("a");
+        link.href = url;
+        link.download = fileName;
+        link.click();
+        link.remove();
+        window.URL.revokeObjectURL(url);
```

Comment:
This is very nit-picky, but I don't think it's the responsibility of this function to revoke the URL

Reply:
You're right! Good point. I'd like to get better at spotting single-responsibility issues, so I appreciate you raising this.